src/adapters/scene.js

```diff
@@ -390,6 +394,27 @@ Scene.prototype.onHashChange = function() {
   };
 };
 
+Scene.prototype.moveMobileBottomUI = function(bottom = 0){
+  if (Device.isMobile()) {
+    let attrib = document.querySelector(".mapboxgl-ctrl-attrib"); // default bottom: 2px
```

Comment:
I understand with all these absolute positionings it's not easy to come with a generic solution, but here there's a lot of code duplication and a hard coupling between CSS and JS (the default bottom), which is always a bad idea. There are at least 2 ways to make this better:
1. see if it's possible to use another CSS property than `bottom`, to be able to apply the translation without having to know the initial value. It's common to use `transform: translateY([value]px)` for that. And as every CSS 3D transform, it has the added benefit of being hardware-accelerated, so it's probably better for overall perf.
2. creating a function taking the selector and new bottom as arguments, so the logic is written once and you have only 4 calls… that you can put in a loop so you just have to declare an array of selectors somewhere.
```js
function moveMobileBottomUI(bottom = 0) {
  const uiControls = ['.mapboxgl-ctrl-attrib', '.map_control__scale', '.mapboxgl-ctrl-geolocate', '.direction_shortcut'];
  uiControls.forEach(uiControl => { translateUIControl(uiControl, bottom); });
}
```